pkg/counter/prefixes.go

```diff
@@ -45,6 +45,44 @@ func NewPrefixLengthCounter(maxUniquePrefixes6, maxUniquePrefixes4 int) *PrefixL
 	}
 }
 
+// This is a bit ugly, but there's not a great way to define an IPNet without
+// parsing strings, etc.
+func createIPNet(ones, bits int) *net.IPNet {
+	return &net.IPNet{
+		Mask: net.CIDRMask(ones, bits),
+	}
+}
+
+type prefixTracker interface {
+	// GetMaxPrefixLengths returns the maximum number of unique prefix
+	// lengths that may be supported in the implementation.
+	GetMaxPrefixLengths(ipv6 bool) int
```

Comment:
Way out of scope but this could always return both ipv4 and ipv6 lengths, and the caller selects what they need/want.